pkg/build/api/validation/validation.go

```diff
@@ -44,11 +44,19 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) fielderrors.ValidationErr
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("namespace", config.Namespace, "namespace must be a valid subdomain"))
 	}
 	allErrs = append(allErrs, validation.ValidateLabels(config.Labels, "labels")...)
-	for i := range config.Triggers {
-		allErrs = append(allErrs, validateTrigger(&config.Triggers[i]).PrefixIndex(i).Prefix("triggers")...)
+	// allow only one ImageChangeTrigger for now
+	ictCount := 0
+	for i, trg := range config.Triggers {
+		allErrs = append(allErrs, validateTrigger(&trg).PrefixIndex(i).Prefix("triggers")...)
+		if trg.Type == buildapi.ImageChangeBuildTriggerType {
+			ictCount++
```

Comment:
A deployment can't trigger on different image changes? What if I have a deployment that has two containers, and I want a new deployment when either of them changes?